esmvalcore/preprocessor/_time.py

```diff
@@ -442,20 +446,41 @@ def anomalies(cube, period):
     ref_coord = _get_period_coord(reference, period)
 
     data = cube.core_data()
-    cube_time = cube.coord('time')
     ref = {}
     for ref_slice in reference.slices_over(ref_coord):
         ref[ref_slice.coord(ref_coord).points[0]] = da.ravel(
             ref_slice.core_data())
     cube_coord_dim = cube.coord_dims(cube_coord)[0]
-    for i in range(cube_time.shape[0]):
-        time = cube_time.points[i]
-        indexes = cube_time.points == time
+    for i in range(cube.coord('time').shape[0]):
+        indexes = cube.coord('time').points == cube.coord('time').points[i]
         indexes = iris.util.broadcast_to_shape(indexes, data.shape,
                                                (cube_coord_dim, ))
         data[indexes] = data[indexes] - ref[cube_coord.points[i]]
 
     cube = cube.copy(data)
+
+    # standardize the results if requested
+    if standardize:
+        cube_stddev = climate_statistics(cube,
+                                         operator='std_dev',
+                                         period=period)
+        if not cube.ndim == cube_stddev.ndim:
+            raise ValueError(
+                "Inconsistent dimensions between anomaly cube and "
+                f"standard deviation cube: {cube.ndim};{cube_stddev.ndim}"
+            )
+        ratio = [i / j for i, j in
```

Comment:
You're only using the first item of `ratio`, so I think there is no need to compute the other items.

Note that you do not have to assume that the first dimension is the time dimension. You can also do something like
```python
tdim = cube.coord_dims('time')[0]
reps = cube.shape[tdim] / cube_stddev.shape[tdim]
weights = da.concatenate([cube_stddev.core_data() for _ in range(ratio)], axis=tdim)
cube.data = cube.core_data() / weights